condacolab.py

```diff
@@ -18,13 +18,20 @@
 from pathlib import Path
 from subprocess import check_output, run, PIPE, STDOUT
 from textwrap import dedent
-from typing import Dict, AnyStr
+from typing import Dict, AnyStr, Iterable
 from urllib.request import urlopen
+from urllib.error import HTTPError
 from distutils.spawn import find_executable
 from IPython.display import display
 
 from IPython import get_ipython
 
+try:
+    from ruamel.yaml import YAML
+    from ruamel.yaml.comments import CommentedMap
+except ImportError as e:
+    raise RuntimeError("Could not find ruamel.yaml, plese install using `!pip install ruamel.yaml`!") from e
+
 try:
     import ipywidgets as widgets
```

Comment:
Two things:

1. Sort dependencies and separate stdlib from 3rd party
2. Add 3rd party packages to `pyproject.toml` dependencies to ensure everything is available in case Google changes the bundled packages.